### ISSUE TITLE
Key events involving the control key don't work as expected

### ISSUE THREAD

**Issue by paulrouget** [servo/servo#17146]
We realized that no keybindings on Windows work (the "reload"/Ctrl-R binding doesn't work for example).

This is because Glutin sends 2 things:
- the character matching the keyboard layout (`ReceivedCharacter(ch)`)
- the key code and modifiers (`KeyboardInput`)

But the character that is sent when the Ctrl key is pressed is a different unicode character.
For exemple, Ctrl-L will send the ["Form Feed" `\u{c}` character](https://en.wikipedia.org/wiki/Page_break#Form_feed), not L.

A possible approach is to map the key code to the character. This is what is done in the DOM code here: https://github.com/servo/servo/blob/985050d3645382d5c2cc6bd6819ab05f9b843331/components/script/dom/keyboardevent.rs#L169

But this is wrong because the event's `key` property is not localized.

Another approach is to map the combo unicode character to its original key. For example:

```
'\u{c}' => 'l',
'\u{12}' => 'r',
…
```

Both the ports/glutin/window.rs and components/script/dom/keyboardevent.rs need this.

**Comment by jklepatch** [comment]
I am working on this

**Comment by atouchet** [comment]
Is #16459 related to this issue?

**Comment by paulrouget** [comment]
> Is #16459 related to this issue?

Very likely, yes.

**Comment by atouchet** [comment]
@paulrouget do you know what the current situation is regarding this issue? As far as I can tell control key input still isn't working properly after the Glutin update.

**Comment by paulrouget** [comment]
We need to investigate see if that needs to be fixed in glutin or in servo.

**Comment by atouchet** [comment]
@paulrouget I've been testing your PR for this and it works somewhat but there are still some definite issues. On first launch using Ctrl + L to open the URL dialog works consistently but subsequent attempts at doing this (ex. load servo.org homepage -> press Ctrl + L to navigate to new page -> press Ctrl + L again to navigate to another new page) do not bring up the dialog again. It seems to work only once and then rarely or not at all afterwards. Should I file a new issue for this behaviour?

**Comment by paulrouget** [comment]
> Should I file a new issue for this behaviour?

Yes please.

Thanks!
